web/app.py

```diff
@@ -0,0 +1,256 @@
+# ============================================
+# FaceAuth Web Application Backend
+# File: app.py
+# Authors: Diogo Azevedo and Letícia Loureiro
+# Date: 2025-04-23
+#
+# Description:
+# This Flask application powers the FaceAuth web platform.
+# It handles:
+# - Manual and facial login
+# - User management (create, remove)
+# - Session management
+# - Real-time face recognition
+# - Embedding generation and storage
+# ============================================
```

Comment:
This script is too large, it is trying to do everything at once - auth, user management, face recognition, and data storage. It should be split into smaller modules.

Also, the API responses are inconsistent - some return plain text, others return JSON with different structures. Standardize all responses to use a consistent JSON format with success/error flags and meaningful status codes so the frontend knows how to handle them.